### PR TITLE
Create new PR with losses and box list ops

### PR DESCRIPTION
**Pull request by kmindspark**
# Description

> :memo: Please include a summary of the change. 
>  
> * Please also include relevant motivation and context.  
> * List any dependencies that are required for this change.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [ ] TensorFlow 2 migration
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] A new research paper code implementation
- [ ] Other (Specify)

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**:

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [ ] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [ ] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ ] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: research/object_detection/utils/ops.py [hunk 1]
```diff
@@ -1134,3 +1134,66 @@ def decode_image(tensor_dict):
...
+                                                      x2I > x1I) else 0.0
+
+    x1C = tf.minimum(pred_xmin, gt_xmin)
+    x2C = tf.maximum(pred_xmax, gt_xmax)
+    y1C = tf.minimum(pred_ymin, gt_ymin)
+    y2C = tf.maximum(pred_ymax, gt_ymax)
+    hull_area = (y2C - y1C) * (x2C - x1C)
+
+    union_area = gt_area + pred_area - intersection_area
+    IoU = intersection_area/union_area if union_area != 0.0 else 1.0
```
Comment (tombstone) [inline research/object_detection/utils/ops.py:1181]: nit: lets keep this consisted with box_list_ops implementation here: https://github.com/tensorflow/models/blob/master/research/object_detection/core/box_list_ops.py#L302

File: research/object_detection/utils/ops.py [hunk 2]
```diff
@@ -1134,3 +1134,66 @@ def decode_image(tensor_dict):
...
+    gIoU = IoU - (hull_area - union_area) / hull_area if hull_area != 0.0 else IoU
```
Comment (tombstone) [inline research/object_detection/utils/ops.py:1182]: tf.where(hull_area>0.0, Iou - (hull_Area - union_area) / hull_area, iou)

File: research/object_detection/utils/ops.py [hunk 3]
```diff
@@ -1134,3 +1134,66 @@ def decode_image(tensor_dict):
...
+
+    return gIoU
+
+  return shape_utils.static_or_dynamic_map_fn(two_boxes_giou, [boxes1, boxes2])
+
+def cy_cx_h_w_to_ymin_xmin_ymax_xmax(input_tensor):
```
Comment (tombstone) [inline research/object_detection/utils/ops.py:1188]: nit: rename to `center_to_corner_coordinate`

File: research/object_detection/utils/ops_test.py [hunk 4]
```diff
@@ -1630,8 +1630,47 @@ def graph_fn():
     self.assertAllClose(expected_gathered_tensor, gathered_tensor_np)
 
 
+class TestGIoU(test_case.TestCase):
 
+  def test_giou(self):
```
Comment (tombstone) [inline research/object_detection/utils/ops_test.py:1635]: I think you can expand the tests to be more comprehensive. Examples - a) non overlapping far boxes, b) non overlapping near boxes, c) overlapping boxes where l1/l2 remains same but giou changes with iou

**Comment by kmindspark** [comment]
Broken

**Pull request closed without merging**